src/utils.ts

```diff
@@ -171,7 +171,7 @@ export async function getMessagesByUserIn(client: MatrixClient, sender: string,
     }
 }
 
-export async function replaceRoomIdsWithPills(client: MatrixClient, text: string, roomIds: string[] | string, msgtype: MessageType = "m.text"): Promise<TextualMessageEventContent> {
+export async function replaceRoomIdsWithPills(client: MatrixClient, logProxy: ILogProxy, text: string, roomIds: string[] | string, msgtype: MessageType = "m.text"): Promise<TextualMessageEventContent> {
```

Comment:
it feels a little bad to be feeding `ILogProxy` in to stuff like this. it's a bit against single responsibility protocol; I'd maybe prefer being able to tell the calling scope what rooms failed to resolve so the calling scope can handle logging

Reply:
in this specific instance we could replace `client: MatrixClient, logProxy: ILogProxy` with a `Mjolnir` but that also feels messy

Reply:
My apologies, I didn't see your comment.

I think that passing a `mjolnir`  arg makes sense. This is pretty much what we're doing everywhere else.